### PR TITLE
P2: Better error message when trying to delete a P2 hub with spaces

### PR DESCRIPTION
**Pull request by naxoc**
#### Changes proposed in this Pull Request

A recent update to the API will refuse to delete a P2 hub if it has spaces. See `D63635-code`.

The error message that is displayed directly from the 403 from the API does not hang out on the screen for very long, so this is an error message that is there until the user dismisses it. It has an error explication and a link for the user to go fix the problem.

#### Testing instructions

* On a P2 hub, go to `http://calypso.localhost:3000/settings/delete-site/[your-site]`.
* Try deleting the site by clicking "Delete Site" - you should get an error in the upper right corner that stays and only goes away when dismissed.

##### Before
before

##### After
after

### CODE REVIEW

File: client/state/sites/actions.js [hunk 1]
```diff
@@ -219,6 +221,24 @@ export function deleteSite( siteId ) {
 					);
 					return;
 				}
+				if ( error.error === 'p2-hub-has-spaces' ) {
+					const hubId = getP2HubBlogId( getState(), siteId );
+					const hubUrl = getSiteUrl( getState(), hubId );
+					dispatch(
+						errorNotice(
+							translate(
+								'Your P2 has spaces. You must delete all spaces before you can delete this P2'
```
Comment (annemirasol) [inline client/state/sites/actions.js]: The second sentence's period,  maybe?

**Review by annemirasol — APPROVED** [review]
As described! 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6146731

Thank you @naxoc for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6146731).

**Pull request merged**